client.go

```diff
@@ -27,13 +27,21 @@ type AutoNATError struct {
 	Text   string
 }
 
+// GetAddrs is a function that returns the addresses to dial back
+type GetAddrs func() []ma.Multiaddr
+
 // NewAutoNATClient creates a fresh instance of an AutoNATClient
-func NewAutoNATClient(h host.Host) AutoNATClient {
-	return &client{h: h}
+func NewAutoNATClient(h host.Host, ga ...GetAddrs) AutoNATClient {
+	getAddrs := h.Addrs
+	if len(ga) > 0 {
+		getAddrs = ga[0]
```

Comment:
Could we get test coverage for this?

Reply:
it's covered in https://github.com/libp2p/go-libp2p-autonat-svc/pull/1 which has an actual service implementation.
It's also implicitly covered in `autonat_test` which uses it internally.